src/common/useIdWithFallback.ts

```diff
@@ -3,16 +3,18 @@
 
 import * as React from "react"
 
+const fallback = () => Math.random().toString(36).substring(2)
+
 /**
  * A function to obtain a unique ID. If react is available, this
  * is just a wrapper for React.useId(), meaning the ID will be
  * consistent across SSR and CSR. Otherwise, it will be a random and
  * unique ID on every call.
  */
 export const useIdWithFallback = () => {
-  try {
-    return React.useId()
-  } catch (e) {
-    return Math.random().toString(36).substring(2)
-  }
+  const useId =
+    (Object.entries(React).find(([key]) => key === "useId")?.[1] as
```

Comment:
Well shoot, you were faster than me with the PR :D I wrote an explaining comment yesterday which I would like to add, considering how hacky this is. Also, we want to ask for a couple more opinions internally before we move on with this, since working behind webpack's back might have some unwanted consequences.

```suggestion
    // This is just a hacky way of optionally importing useId. If
    // it's imported the regular way, or we even attempt to read
    // the `useId` property of React, webpack seems to throw an
    // error. Since `useId` isn't a hard dependency, we're fine
    // with it not existing and this prevents the error from
    // occurring.
    (Object.entries(React).find(([key]) => key === "useId")?.[1] as
```